### PR TITLE
Add bench for fusion

### PR DESCRIPTION
**Pull request by asi1024**
```
[ 33.33%] ··· Running cupy.bench_fusion.Fusion.time_fusion_add_10_times                                                                                                                                   ok
[ 33.33%] ···· 
               ============== ==========
                fusion_param            
               -------------- ----------
                  enabled      89.89μs  
                  disabled     461.94μs 
               ============== ==========

[ 66.67%] ··· Running cupy.bench_fusion.Fusion.time_fusion_loops                                                                                                                                          ok
[ 66.67%] ···· 
               ============== ==========
                fusion_param            
               -------------- ----------
                  enabled      170.15μs 
                  disabled     14.21ms  
               ============== ==========

[100.00%] ··· Running cupy.bench_fusion.Fusion.time_fusion_lstm_grad_grad                                                                                                                                 ok
[100.00%] ···· 
               ============== ==========
                fusion_param            
               -------------- ----------
                  enabled      616.33μs 
                  disabled      5.06ms  
               ============== ==========
```

### CODE REVIEW

File: benchmarks/cupy/bench_fusion.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                args.append(uniform(0, 200, shape, cupy.float32))
+        # TODO (imanishi)
+        # fused_func.compile(*args)
+        fused_func(*args)
+        return wrapper
+    return decorator
+
+
+@sync
+@parameterize([('fusion_param', ['enabled', 'disabled'])])
```
Comment (kmaehashi) [inline benchmarks/cupy/bench_fusion.py]: `fusion_param` sounds like arguments to `cupy.fuse`.
How about `fusion_mode`? (like [backend mode](https://github.com/chainer/chainer-benchmark/blob/master/benchmarks/utils/backend.py#L14) in chainer-benchmark?)

File: benchmarks/cupy/bench_fusion.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+class Fusion(Benchmark):
+    def setup(self, fusion_enabled):
+        self.x = cupy.arange(240000.0).reshape(400, 600)
```
Comment (kmaehashi) [inline benchmarks/cupy/bench_fusion.py]: dtype of `x` will be `float64`. Is this expected?

File: benchmarks/cupy/bench_fusion.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        fused_func = cupy.fuse(func)
+        @functools.wraps(func)
+        def wrapper(self, fusion_param):
+            if fusion_param == 'enabled':
+                fused_func(*args)
+            elif fusion_param == 'disabled':
+                func(*args)
+            elif fusion_param == 'compile':
+                raise NotImplementedError
+                # TODO (imanishi)
```
Comment (kmaehashi) [inline benchmarks/cupy/bench_fusion.py]: Conventionally no spaces between TODO and name.
(This is not a documented rule but culture coming of Google's Style Guide: https://google.github.io/styleguide/pyguide.html#TODO_Comments)

File: benchmarks/cupy/bench_fusion.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                # fused_func._memo = {}
+                # fused_func.compile(*args)
+            else:
+                raise ValueError('Invalid parameter')
+
+        uniform = cupy.random.uniform
+        args = []
+        if shape_args:
+            for name in inspect.getargspec(func).args:
```
Comment (kmaehashi) [inline benchmarks/cupy/bench_fusion.py]: `assert len(shapes_dict) == 0` here.

File: benchmarks/cupy/bench_fusion.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        # fused_func.compile(*args)
+        fused_func(*args)
+        return wrapper
+    return decorator
+
+
+@sync
+@parameterize([('fusion_param', ['enabled', 'disabled'])])
+class Fusion(Benchmark):
+    def setup(self, fusion_enabled):
```
Comment (kmaehashi) [inline benchmarks/cupy/bench_fusion.py]: `fusion_param` (or `fusion_mode`)

**Review by kmaehashi — CHANGES_REQUESTED** [review]
Thanks! LGTM except comments.

File: benchmarks/cupy/bench_fusion.py (hunk 1 above)
Comment (asi1024) [reply benchmarks/cupy/bench_fusion.py]: Fixed.

File: benchmarks/cupy/bench_fusion.py (hunk 2 above)
Comment (asi1024) [reply benchmarks/cupy/bench_fusion.py]: `self.x` is not referred to, so I removed it.

File: benchmarks/cupy/bench_fusion.py (hunk 3 above)
Comment (asi1024) [reply benchmarks/cupy/bench_fusion.py]: Fixed.

File: benchmarks/cupy/bench_fusion.py (hunk 4 above)
Comment (asi1024) [reply benchmarks/cupy/bench_fusion.py]: Added.

File: benchmarks/cupy/bench_fusion.py (hunk 5 above)
Comment (asi1024) [reply benchmarks/cupy/bench_fusion.py]: Fixed.

**Review by kmaehashi — APPROVED** [review]

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**